pyvista/core/utilities/misc.py

```diff
@@ -317,3 +319,27 @@ def _reciprocal(x: ArrayLike[float], tol: float = 1e-8) -> NumpyArray[float]:
     x[~zero] = np.reciprocal(x[~zero])
     x[zero] = 0
     return x
+
+
+class classproperty(property):
```

Comment:
I think we should make this private.  We don't want downstream packages using these things.  If we dont like the `@_classproperty` usage, it would be better if we had a `_misc` module.